### PR TITLE
Use `Collection.partition(by:)` and remove `Collection.split(_:)`

### PR DESCRIPTION
**Pull request by norio-nomura**
Fix [SR-6523](https://bugs.swift.org/browse/SR-6523)

### CODE REVIEW

File: Sources/Xcodeproj/pbxproj().swift [hunk 1]
```diff
@@ -296,7 +296,7 @@ func xcodeProject(
         return sourcesGroup ?? srcPathsToGroups[targets[0].sources.root]
     }
 
-    let (rootModules, testModules) = graph.rootPackages[0].targets.split{ $0.type != .test }
+    let (rootModules, testModules) = graph.rootPackages[0].targets.classify{ $0.type != .test }
```
Comment (aciidb0mb3r) [inline Sources/Xcodeproj/pbxproj().swift]: If this is the only client, I wonder if we even need this extension.

File: Sources/Utility/CollectionExtensions.swift [hunk 2]
```diff
@@ -34,12 +34,12 @@ extension Collection where Iterator.Element : Equatable {
...
 
 extension Collection {
-    /// Splits into two arrays, first containing elements which matching the predicate
+    /// Classifies into two arrays, first containing elements which matching the predicate
     /// and other containting elements not matching the predicate.
     ///
     /// - Parameter isMatching: The predicate to apply.
     /// - Returns: Tuple of split arrays.
-    public func split(_ isMatching: (Iterator.Element) -> Bool) -> ([Iterator.Element], [Iterator.Element]) {
+    public func classify(_ isMatching: (Iterator.Element) -> Bool) -> ([Iterator.Element], [Iterator.Element]) {
```
Comment (segiddins) [inline Sources/Utility/CollectionExtensions.swift]: I believe Ruby spells this particular method as `partition`

Comment (aciidb0mb3r) [reply Sources/Utility/CollectionExtensions.swift]: Sounds good to me!

**Comment by norio-nomura** [comment]
It seems `MutableCollection.partition(by:)` exists as mutating func. 🤔
https://developer.apple.com/documentation/swift/mutablecollection/2905358-partition

**Comment by aciidb0mb3r** [comment]
Thats great, can we use the stdlib method then?

**Comment by norio-nomura** [comment]
Updated to use stdlib method.

**Comment by aciidb0mb3r** [comment]
Perfect, thanks!

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by norio-nomura** [comment]
Thanks! 🙏

**Pull request merged**